src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -216,6 +216,86 @@ g.test('mipLevelCount,bound_check,bigger_than_integer_bit_width')
     });
   });
 
+g.test('sampleCount,various_sampleCount_with_all_formats')
+  .desc(`Test texture creation with various (valid or invalid) sample count and all formats`)
+  .subcases(() =>
+    params()
+      .combine(poptions('sampleCount', [1, 2, 4, 8, 16, 32, 256]))
+      .combine(poptions('format', kAllTextureFormats))
+  )
+  .fn(async t => {
+    const { sampleCount, format } = t.params;
+    const descriptor = {
+      size: [32, 32, 1],
+      sampleCount,
+      format,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
+
+    const success =
+      sampleCount === 1 || (sampleCount === 4 && format in kUncompressedTextureFormats);
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
+  });
+
+g.test('sampleCount,valid_sampleCount_with_other_parameter_varies')
+  .desc(
+    `Test texture creation with valid sample count when dimensions, arrayLayerCount, mipLevelCount, format, and usage varies.`
+  )
+  .subcases(() =>
+    params()
+      .combine(poptions('sampleCount', [1, 4]))
+      .combine(poptions('dimension', ['1d', '2d', '3d'] as const))
+      .combine(poptions('arrayLayerCount', [1, 2]))
+      .unless(({ arrayLayerCount, dimension }) => arrayLayerCount === 2 && dimension !== '2d')
+      .combine(poptions('mipLevelCount', [1, 2]))
+      .combine(poptions('format', kAllTextureFormats))
+      .combine(poptions('usage', kTextureUsages))
+      .unless(({ usage, format }) => {
+        const info = kAllTextureFormatInfo[format];
+        return (
+          ((usage & GPUConst.TextureUsage.RENDER_ATTACHMENT) !== 0 && !info.renderable) ||
+          ((usage & GPUConst.TextureUsage.STORAGE) !== 0 && !info.storage)
+        );
+      })
+  )
+  .fn(async t => {
+    const { sampleCount, dimension, format, mipLevelCount, arrayLayerCount, usage } = t.params;
+
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
+
+    const size =
+      dimension === '1d'
+        ? [32, 1, 1]
+        : dimension === '2d'
+        ? [32, 32, arrayLayerCount]
+        : [32, 32, 32];
+    const descriptor = {
+      size,
+      mipLevelCount,
+      sampleCount,
+      dimension,
+      format,
+      usage,
+    };
+
+    const success =
+      sampleCount === 1 ||
+      (sampleCount === 4 &&
+        dimension === '2d' &&
+        format in kUncompressedTextureFormats &&
+        mipLevelCount === 1 &&
+        arrayLayerCount === 1 &&
+        (usage & GPUConst.TextureUsage.STORAGE) === 0);
+
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
+  });
+
 g.test('sampleCount')
```

Comment:
I think this entire test is subsumed by your more thorough tests. If you agree, let's go ahead and delete it.

Reply:
Agreed. This PR is not ready at that time, so I didn't remove the original code. 
Done.